frontend/src/store/newUserComparatorState.ts

```diff
@@ -0,0 +1,180 @@
+import { ref } from 'vue';
+import type {
+    FamApplicationUserRoleAssignmentGet,
+    FamUserRoleAssignmentCreate,
+} from 'fam-app-acsctl-api/model';
+import type {
+    FamAccessControlPrivilegeCreateRequest,
+    FamAccessControlPrivilegeGetResponse,
+    FamAppAdminCreateRequest,
+    FamAppAdminGetResponse,
+} from 'fam-admin-mgmt-api/model';
+import type { TabKey } from '@/enum/TabEnum';
+
+interface NewUsersValue {
+    userAccess: FamUserRoleAssignmentCreate[];
+    adminAccess: FamAppAdminCreateRequest[];
+    delegatedAdminAccess: FamAccessControlPrivilegeCreateRequest[];
+}
+
+// Initialize an array to store new users
+const newUsers = ref<NewUsersValue>({
+    userAccess: [],
+    adminAccess: [],
+    delegatedAdminAccess: [],
+});
+
+export const clearNewUserTag = () =>
+    (newUsers.value = {
+        userAccess: [],
+        adminAccess: [],
+        delegatedAdminAccess: [],
+    });
+
+export const setNewUsers = (
+    newItem:
+        | FamUserRoleAssignmentCreate
+        | FamAppAdminCreateRequest
+        | FamAccessControlPrivilegeCreateRequest,
+    arrayToUpdate: TabKey
+) => {
+    newUsers.value = {
+        ...newUsers.value,
+        [arrayToUpdate]: [...newUsers.value[arrayToUpdate], newItem],
+    };
+};
+
+export const highlightNewUserRow = (rowData: any) => {
+    console.log('row', rowData)
+    if(rowData.isNewUser) {
+        return {
+            'background-color': '#C2E0FF',
+            'box-shadow': 'inset 0 0 0 0.063rem #85C2FF'
+        }
+    }
+}
+
+/**
+ * Compares a list of user role assignments and identifies new users.
+ * @param userRoleAssignments An array of user role assignments.
+ * @returns A sorted array of user role assignments with an additional property indicating if the user is new.
+ */
+export const compareUserTable = (
+    userRoleAssignments: FamApplicationUserRoleAssignmentGet[] = []
```

Comment:
I wonder instead of creating these compare method, maybe we can modify the fetch methods. So before entering each route, we'll fetch the user access data. And when do the sorting here, found the new user access record, and put it on the top. 

<img width="1141" alt="Screen Shot 2024-06-06 at 11 58 17 AM" src="https://github.com/bcgov/nr-forests-access-management/assets/23131735/22498b8f-c344-4045-84df-0b8dc7369afc">

Reply:
and instead of comparing user_name and etc, we could compare the primary key that get from the return of the POST request, so you should store the result of the POST request, and only do that for successful cases. 
<img width="1074" alt="Screen Shot 2024-06-06 at 12 36 12 PM" src="https://github.com/bcgov/nr-forests-access-management/assets/23131735/f96b06bc-d47a-4819-ac32-f1f1d974879f">
<img width="1112" alt="Screen Shot 2024-06-06 at 12 29 48 PM" src="https://github.com/bcgov/nr-forests-access-management/assets/23131735/1c343430-bf7d-4737-b4cb-b4169842d8fc">
<img width="1094" alt="Screen Shot 2024-06-06 at 12 36 44 PM" src="https://github.com/bcgov/nr-forests-access-management/assets/23131735/484c455d-690d-4d6d-ae4c-c50cb8c701b1">
<img width="1309" alt="Screen Shot 2024-06-06 at 12 31 20 PM" src="https://github.com/bcgov/nr-forests-access-management/assets/23131735/03c7a1ca-ec47-420a-9d55-eef4364ba712">

<img width="1110" alt="Screen Shot 2024-06-06 at 12 37 28 PM" src="https://github.com/bcgov/nr-forests-access-management/assets/23131735/ecafab34-04d0-4b6d-a472-02d9db4940f2">
<img width="1195" alt="Screen Shot 2024-06-06 at 12 30 51 PM" src="https://github.com/bcgov/nr-forests-access-management/assets/23131735/535f4f4f-c073-4453-bfba-46f6f9028c1f">

Reply:
sounds good, I'll follow your suggestion